packages/grid/_modules_/grid/hooks/features/columns/gridColumnsUtils.ts

```diff
@@ -100,6 +100,17 @@ export const getGridColDef = (
   if (!type) {
     return columnTypes[DEFAULT_GRID_COL_TYPE_KEY];
   }
+
+  if (process.env.NODE_ENV !== 'production' && !columnTypes[type]) {
```

Comment:
We have the convention of only warning once.

https://github.com/mui-org/material-ui-x/blob/d5d6bc6d32b5b6a43a4c939a6831c00e974563e2/packages/grid/_modules_/grid/utils/createSelector.ts#L40-L41